webapp/components/Editor/ContentViewer.vue

```diff
@@ -31,6 +31,9 @@ export default {
       }),
     }
   },
+  beforeUpdate() {
+    this.editor.setContent(this.content)
```

Comment:
> [<img alt="roschaefer" height="40" width="40" align="left" src="https://avatars1.githubusercontent.com/u/38261864?s=88&v=4">](https://github.com/roschaefer) **Authored by [roschaefer](https://github.com/roschaefer)**
_<time datetime="2019-10-15T22:09:30Z" title="Wednesday, October 16th 2019, 12:09:30 am +02:00">Oct 16, 2019</time>_

---
> **Outdated (history rewrite)** - original diff
---
```diff
@@ -31,6 +31,9 @@ export default {
       }),
     }
   },
+  beforeUpdate() {
+    this.editor.setContent(this.content)
```


Could you give me an idea what the effect of this is? :thinking:

Reply:
> [<img alt="alina-beck" height="40" width="40" align="left" src="https://avatars1.githubusercontent.com/u/38261864?s=88&v=4">](https://github.com/alina-beck) **Authored by [alina-beck](https://github.com/alina-beck)**
_<time datetime="2019-10-16T07:52:59Z" title="Wednesday, October 16th 2019, 9:52:59 am +02:00">Oct 16, 2019</time>_

---


Yep! So, we're using the `tiptap editor` to display comments. The problem was that when the `content` prop changed the `ContentViewer` didn't update so the comment would always stay in its initial state (i.e. it could not be collapsed or expanded).

To solve this I had to a) use the `content` prop within the `<template>` to trigger an update at all – so I set it as a `key` and b) update the editor with the new `content` (which is happening above).

This was not necessary before because we were using three different instances of the `ContentViewer`.

Reply:
> [<img alt="roschaefer" height="40" width="40" align="left" src="https://avatars1.githubusercontent.com/u/38261864?s=88&v=4">](https://github.com/roschaefer) **Authored by [roschaefer](https://github.com/roschaefer)**
_<time datetime="2019-10-16T10:12:49Z" title="Wednesday, October 16th 2019, 12:12:49 pm +02:00">Oct 16, 2019</time>_

---


Thanks, sounds more like the VueJS way. :+1: